test/tests/CMakeLists.txt

```diff
@@ -1,198 +1,43 @@
+   "${CMAKE_CURRENT_SOURCE_DIR}/CLITests.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/CryptTests.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/Endianness.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/EnumMapTest.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/FormattingTests.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/ImageImporterTests.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/IniReaderTest.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/IniWriterTest.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/LanguagePackTest.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/Localisation.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/MultiLaunch.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/Pathfinding.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/Platform.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/PlayTests.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/ReplayTests.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/RideRatings.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/S6ImportExportTests.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/SawyerCodingTest.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/TestData.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/TestData.h"
+   "${CMAKE_CURRENT_SOURCE_DIR}/tests.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/TileElements.cpp"
+   "${CMAKE_CURRENT_SOURCE_DIR}/TileElementsView.cpp")
+
+add_executable(OpenRCT2Tests ${test_files})
+target_link_libraries(OpenRCT2Tests GTest::gtest GTest::gtest_main libopenrct2)
+target_include_directories(OpenRCT2Tests PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/../../src")
+set_target_properties(OpenRCT2Tests PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+gtest_discover_tests(OpenRCT2Tests WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
 
-include_directories(SYSTEM ${GTEST_INCLUDE_DIRS})
-include_directories("${ROOT_DIR}/src")
-include_directories(SYSTEM ${SDL2_INCLUDE_DIRS})
 
-# Some most common files required in tests
-set(COMMON_TEST_SOURCES
-    "${ROOT_DIR}/src/openrct2/core/Console.cpp"
-    "${ROOT_DIR}/src/openrct2/core/Diagnostics.cpp"
-    "${ROOT_DIR}/src/openrct2/core/Guard.cpp"
-    "${ROOT_DIR}/src/openrct2/core/String.cpp"
-    "${ROOT_DIR}/src/openrct2/Diagnostic.cpp"
-    "${ROOT_DIR}/src/openrct2/localisation/ConversionTables.cpp"
-    "${ROOT_DIR}/src/openrct2/localisation/Convert.cpp"
-    "${ROOT_DIR}/src/openrct2/localisation/FormatCodes.cpp"
-    "${ROOT_DIR}/src/openrct2/localisation/UTF8.cpp"
-    "${ROOT_DIR}/src/openrct2/util/Util.cpp"
-    "${ROOT_DIR}/src/openrct2/Version.cpp"
-    )
-
-# Create a re-usable library to save some compilation time
-add_library(test-common STATIC ${COMMON_TEST_SOURCES})
-set_target_properties(test-common PROPERTIES COMPILE_DEFINITIONS DISABLE_HTTP)
-
-# Setup testdata. It should be fine here, as the only way to reach here is by explicitly requesting tests.
-execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink "${CMAKE_CURRENT_LIST_DIR}/testdata" "${CMAKE_CURRENT_BINARY_DIR}/testdata")
-install(CODE "execute_process(COMMAND \"${CMAKE_COMMAND}\" -E create_symlink \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_DATADIR}/openrct2\" \"${CMAKE_CURRENT_BINARY_DIR}/data\")")
```

Comment:
It probably doesn't find the data because of this

Reply:
whats really fun is that it does find the data when i run it locally :)